File: dotnet/Lexy.Poc/Lexy.Poc.Core/Parser/IParserContext.cs
using Lexy.Poc.Core.Language;

namespace Lexy.Poc.Core.Parser
{
    public interface IParserContext
    {
        IParserLogger Logger { get; }

        Line CurrentLine { get; }

        Nodes Nodes { get; }

        void ProcessNode(IRootNode node);

        bool ProcessLine();

        TokenValidator ValidateTokens<T>();
        TokenValidator ValidateTokens(string name);
    }
}

File: typescipt/existing/src/parser/iParserContext.ts


namespace Lexy.Compiler.Parser;

public interface IParserContext
{
   IParserLogger Logger { get; }

   Line CurrentLine { get; }

   ISourceCodeDocument SourceCode { get; }
   RootNodeList Nodes { get; }
   SourceCodeNode RootNode { get; }

   void AddFileIncluded(string fileName);
   bool IsFileIncluded(string fileName);
}
